src/briefcase/exceptions.py

```diff
@@ -137,11 +137,10 @@ def __init__(self, error_msg):
         super().__init__(msg=error_msg, skip_logfile=True)
 
 
-class TemplateUnsupportedVersion(BriefcaseCommandError):
-    def __init__(self, briefcase_version):
-        self.briefcase_version = briefcase_version
+class InvalidTemplateBranch(BriefcaseCommandError):
+    def __init__(self, template_repo, branch):
```

Comment:
It's helpful for exceptions to keep their properties, so that they can be interrogated programmatically.
```suggestion
    def __init__(self, template_repo, branch):
        self.template_repo = template_repo
        self.branch = branch
```